### PR TITLE
Documentacion librs

### PR DESCRIPTION
**Pull request by EdVeralli**
documentation added in Lib.rs

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -70,6 +71,18 @@ pub type CircuitOutputType = IndexMap<String, CircuitIOType>;
 
 pub type SimpleProgramVariables = IndexMap<String, Option<CircuitIOType>>;
 
+
+/// Return a Circuit and the Marlin Proof for a given function and it´s inputs.
```
Comment (jrchatruc) [inline src/lib.rs:75]: This should be "Returns the circuit outputs and the marlin proof..."

File: src/lib.rs [hunk 2]
```diff
@@ -155,6 +175,15 @@ fn program_variables(function: &Function<Testnet3>) -> SimpleProgramVariables {
     registers
 }
 
+/// Return a Circuit of a given function and it's variables.
```
Comment (jrchatruc) [inline src/lib.rs:178]: This should be something like "returns a hash map with the circuit outputs"

Comment (EdVeralli) [reply src/lib.rs:178]: It's necessary to include "...of a given function and its variables" at the end of the sentence?

File: src/lib.rs (hunk 1 above)
Comment (EdVeralli) [reply src/lib.rs:75]: ok!

File: src/lib.rs [hunk 3]
```diff
@@ -172,6 +201,23 @@ fn circuit_outputs(
     Ok(circuit_outputs)
 }
 
+/// Analyze user inputs types.
```
[the remark this answers is not in the document]
Comment (EdVeralli) [reply src/lib.rs:204]: ok!

File: src/lib.rs [hunk 4]
```diff
@@ -319,6 +366,25 @@ fn process_inputs(
     Ok(())
 }
 
+/// Analyze output registers and operands.
```
[the remark this answers is not in the document]
Comment (EdVeralli) [reply src/lib.rs:369]: ok!

**Review by ilitteri — APPROVED** [review]
LGTM

**Pull request merged**